packages/contracts-bedrock/contracts/universal/ProxyAdmin.sol

```diff
@@ -140,40 +146,38 @@ contract ProxyAdmin is Owned {
     }
 
     /**
-     * @dev Returns the current implementation of `proxy`.
-     *      This contract must be the admin of `proxy`.
+     * @notice Returns the implementation of the given proxy address.
+     *
+     * @param _proxy Address of the proxy to get the implementation of.
      *
-     * @param proxy The Proxy to return the implementation of.
-     * @return The address of the implementation.
+     * @return Address of the implementation of the proxy.
      */
-    function getProxyImplementation(Proxy proxy) external view returns (address) {
-        ProxyType proxyType = proxyType[address(proxy)];
-
+    function getProxyImplementation(address _proxy) external view returns (address) {
```

Comment:
Changing it from `Proxy` to `address` technically weakens the type safety

Reply:
Sort of, but also the input type is not technically always a `Proxy` since it can also be a `ResolvedDelegateProxy`. In that sense, using the `Proxy` type ends up making this function more confusing.